### PR TITLE
Redesign events

### PR DESCRIPTION
**Pull request by tiagolimpe**
**Feature/Bug description:** ...

**Solution:** 
selection_001
selection_002

**With filters:**
peek 2018-11-19 17-44

**TODO/FIXME:** n/a

### CODE REVIEW

File: backend/handlers/event_collection_handler.py [hunk 1]
```diff
@@ -13,10 +13,32 @@
 from utils import NotAuthorizedException
 from utils import query_paginated
 from utils import to_int
+from datetime import datetime
 from custom_exceptions import QueryException
 
 __all__ = ['EventCollectionHandler']
 
+def get_query_events(filters, user):
```
Comment (raonismaneoto) [inline backend/handlers/event_collection_handler.py]: get filtered events is a better name, isn't it?

File: backend/handlers/event_collection_handler.py [hunk 2]
```diff
@@ -13,10 +13,32 @@
...
+    """Get query of events based on filters by date or not.
+
+    Args:
+        filters: A list of tuples with the name and value of filters to the query
+        Filters by month and year are in end of list
+        user: The current logged user.
+    """
+    if len(filters) > 2:
```
Comment (raonismaneoto) [inline backend/handlers/event_collection_handler.py]: extract this magic number 2 to a well named constant

File: frontend/event/event.css [hunk 3]
```diff
@@ -179,4 +179,120 @@
...
+    }
+
+    .event-position {
+        grid-column-start: 2;
+        grid-column-end: 3;
+    }
+
+    .events-column {
+        display: grid;
+        grid-template-columns: auto;
```
Comment (raonismaneoto) [inline frontend/event/event.css]: i think this is useless

File: frontend/event/event.css [hunk 4]
```diff
@@ -179,4 +179,120 @@
...
+        grid-template-rows: auto;
+    }
+
+    .event-preview-image {
+        max-height: 120px;
```
Comment (raonismaneoto) [inline frontend/event/event.css]: use 'em' in everywhere you're using px

File: frontend/event/eventController.js [hunk 5]
```diff
@@ -2,15 +2,20 @@
...
     var app = angular.module('app');
 
-    app.controller("EventController", function EventController(EventService, $state, $mdDialog, AuthService, $q) {
+    app.controller("EventController", function EventController(EventService, $state, $mdDialog, AuthService, $q, $http) {
         var eventCtrl = this;
         var content = document.getElementById("content");
 
         var moreEvents = true;
         var actualPage = 0;
+        var isAnotherMonth = false;
```
Comment (raonismaneoto) [inline frontend/event/eventController.js]: let

File: frontend/event/eventController.js [hunk 6]
```diff
@@ -79,9 +90,117 @@
...
+                        day: day,
+                        events: eventsByDay[day]
+                    };
+                    eventCtrl.eventsByDay.push(currentValue);
+                });
+            }
+        };
+
+        /**
+         * Loads the years 2017 to 2050 to show in filter by year
```
Comment (raonismaneoto) [inline frontend/event/eventController.js]: get the current year, add some constant quantity and set the limit to that year, it's better.

File: frontend/event/events.html [hunk 7]
```diff
@@ -26,7 +26,7 @@ <h4 class="font-resp create-event">CRIAR EVENTO</h4>
           </md-card>
         </div>
         <load-circle flex add-layout-fill="true" ng-if="eventCtrl.isLoadingEvents"></load-circle>
-        <md-card ng-if="eventCtrl.events.lentgh === 0" style="margin-left: 40px;">
+        <md-card ng-if="eventCtrl.events.length === 0" style="margin-left: 40px;">
```
Comment (raonismaneoto) [inline frontend/event/events.html]: move this style to a css' file

File: backend/handlers/event_collection_handler.py (hunk 1 above)
Comment (raonismaneoto) [reply backend/handlers/event_collection_handler.py]: you gotta test this function very well...

File: frontend/test/specs/event/eventControllerSpec.js [hunk 8]
```diff
@@ -91,66 +101,110 @@
         eventCtrl = createCtrl();
         eventCtrl.showImage = true;
         eventCtrl.events = [];
-        $httpBackend.flush()
     }));
 
-    describe('main() without definy institution_key', function() {
+    describe('main() without definy institution_key', () => {
```
Comment (raonismaneoto) [inline frontend/test/specs/event/eventControllerSpec.js]: define*

File: frontend/test/specs/event/eventControllerSpec.js [hunk 9]
```diff
@@ -91,66 +101,110 @@
...
 
-        beforeEach(function () {
+        it("Should not has an institution_key", () => {
```
Comment (raonismaneoto) [inline frontend/test/specs/event/eventControllerSpec.js]: should not have

File: frontend/test/specs/event/eventControllerSpec.js [hunk 10]
```diff
@@ -91,66 +101,110 @@
...
+    });
 
-            spyOn(eventService, 'getEvents').and.callFake(function () {
-                return {
-                    then: function (callback) {
-                        return callback(requestEvent);
-                    }
-                };
-            });
+    describe('main() definy institution_key', () => {
```
Comment (raonismaneoto) [inline frontend/test/specs/event/eventControllerSpec.js]: define*

File: frontend/test/specs/event/eventControllerSpec.js [hunk 11]
```diff
@@ -91,66 +101,110 @@
...
+
+        beforeEach( () => {
+            state.params.institutionKey = institution.key;
         });
 
-        it("Should call loadMoreEvents", function() {
+        it("Should not has an institution_key", () => {
```
Comment (raonismaneoto) [inline frontend/test/specs/event/eventControllerSpec.js]: have*

File: frontend/test/specs/event/eventControllerSpec.js [hunk 12]
```diff
@@ -91,66 +101,110 @@
...
             eventCtrl.newEvent();
             expect(mdDialog.show).toHaveBeenCalled();
         });
     });
 
-    describe('share()', function() {
+    describe('share()', () => {
 
-        it('should call functions', function() {
+        it('should call functions', () => {
```
Comment (raonismaneoto) [inline frontend/test/specs/event/eventControllerSpec.js]: improve this description. Which functions?

File: frontend/event/eventDialogController.js [hunk 13]
```diff
@@ -51,6 +51,9 @@
 
         dialogCtrl.createInitDate = function createInitDate() {
             dialogCtrl.startTime = dialogCtrl.event.start_time && new Date(dialogCtrl.event.start_time);
+            if(dialogCtrl.isEditing) {
```
Comment (JuliePessoa) [inline frontend/event/eventDialogController.js]: why this was added?

File: frontend/event/events_mobile.html [hunk 14]
```diff
@@ -0,0 +1,69 @@
+<div class="container-resp box">
+  <div flex layout="column">
+    <load-circle flex add-layout-fill="true" ng-if="eventCtrl.isLoadingEvents"></load-circle>
+    <div ng-if="!eventCtrl.isLoadingEvents" class="centralized-event-content">
+      <md-content ng-if="eventCtrl.months.length > 0" layout="row">
```
Comment (JuliePessoa) [inline frontend/event/events_mobile.html]: use layout grid instead of 'layout=row'

File: frontend/event/events_mobile.html [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+        ng-if="eventCtrl.events.length > 0" class="events-grid events-grid-position">
+          <div class="events-day-position event-day-properties">
+            <span>{{events.day}}</span>
+            <p class="event-weekday">
+              {{eventCtrl.test(events.events[0])}}
+              {{events.events[0].start_time | amUtc | amDateFormat:'ddd' }}
+            </p>
+          </div>
+          <div ng-repeat="event in events.events | orderBy:'start_time'" class="event-position">
+            <div ng-click="eventCtrl.goToEvent(event)">
```
Comment (JuliePessoa) [inline frontend/event/events_mobile.html]: this ng-click can be placed in the div above

File: frontend/test/specs/event/eventControllerSpec.js [hunk 16]
```diff
@@ -2,56 +2,60 @@
...
-
-    var GET_EVENTS_URI = '/api/events?page=0&limit=5';
-    var GET_EVENTS_INST_URI = '/api/institutions/'+institution.key+'/events?page=0&limit=5';
+    let // variables to be associated to the injected parameters
+        eventCtrl, scope, httpBackend, rootScope,
+        createCtrl, eventService, messageService, mdDialog, state;
+
+    let // variables to create the test scenario
+        institution = { name: 'Institution', key: '098745' },
+        other_institution = { name: 'Ohter Institution', key: '75368' },
```
Comment (JuliePessoa) [inline frontend/test/specs/event/eventControllerSpec.js]: Other

File: frontend/test/specs/event/eventControllerSpec.js [hunk 17]
```diff
@@ -78,6 +82,8 @@
             };
         });
 
+        spyOn(Utils, 'isMobileScreen').and.returnValue(true);
```
Comment (JuliePessoa) [inline frontend/test/specs/event/eventControllerSpec.js]: create an $onInit function in the controller instated of the main function there.
So you can test it properly.
And test when is mobile screen or not.

File: frontend/test/specs/event/eventControllerSpec.js [hunk 18]
```diff
@@ -91,66 +97,111 @@
         eventCtrl = createCtrl();
         eventCtrl.showImage = true;
         eventCtrl.events = [];
-        $httpBackend.flush()
     }));
 
-    describe('main() without definy institution_key', function() {
+    describe('main() without define institution_key', () => {
```
Comment (JuliePessoa) [inline frontend/test/specs/event/eventControllerSpec.js]: this two describes can be just one. because its testing the same functions.

File: frontend/test/specs/event/eventControllerSpec.js [hunk 19]
```diff
@@ -91,66 +97,111 @@
...
+    });
 
-            spyOn(eventService, 'getEvents').and.callFake(function () {
-                return {
-                    then: function (callback) {
-                        return callback(requestEvent);
-                    }
-                };
-            });
+    describe('main() define institution_key', () => {
```
Comment (JuliePessoa) [inline frontend/test/specs/event/eventControllerSpec.js]: issue above

File: frontend/test/specs/event/eventControllerSpec.js [hunk 20]
```diff
@@ -91,66 +97,111 @@
...
+        });
+
+        it('Should call state.go', () => {
+            eventCtrl.goToEvent(event);
+            expect(state.go).toHaveBeenCalledWith('app.user.event', { eventKey: event.key });
         });
     });
 
-    describe('main() definy institution_key', function() {
+    describe('loadFilteredEvents()', () => {
```
Comment (JuliePessoa) [inline frontend/test/specs/event/eventControllerSpec.js]: test if the `moreEvents, actualPAge, isAnotherMonth' change theis values.

File: frontend/test/specs/event/eventControllerSpec.js [hunk 21]
```diff
@@ -91,66 +97,111 @@
...
-            state.params.institutionKey = institution.key;
-            spyOn(eventService, 'getInstEvents').and.callFake(function () {
+        it('Should call loadMoreEvents', () => {
+            eventCtrl.loadFilteredEvents();
+            expect(eventCtrl.loadMoreEvents).toHaveBeenCalled();
+        });
+    });
+
+    describe('_getEventsByDay()', () => {
+        it('Should populate the eventsByDay array', () => {
```
Comment (JuliePessoa) [inline frontend/test/specs/event/eventControllerSpec.js]: test if they have the correct days

File: frontend/test/specs/event/eventControllerSpec.js [hunk 22]
```diff
@@ -91,66 +97,111 @@
...
+            eventCtrl.events = requestEvent.events;
+            eventCtrl.selectedMonth = months[11];
+            expect(eventCtrl.eventsByDay.length).toEqual(0);
+            eventCtrl._getEventsByDay();
+            expect(eventCtrl.eventsByDay.length).toEqual(4);
+        });
+    });
+
+    describe('getMonths()', () => {
```
Comment (JuliePessoa) [inline frontend/test/specs/event/eventControllerSpec.js]: test if selectMonth and selectedYear are correctly defined

File: frontend/event/eventController.js [hunk 23]
```diff
@@ -79,9 +94,135 @@
...
+            isAnotherMonth = true;
+            eventCtrl.loadMoreEvents();
+        };
+
+        /**
+         * @param {object} event the event that be calculated the range that happens
+         * @returns {array} array with init day and end day in positions 0 and 1, respectively
+         * @private
+         */
+        eventCtrl._getDaysRange = (event) => {
```
Comment (JuliePessoa) [inline frontend/event/eventController.js]: teste

File: frontend/event/eventController.js [hunk 24]
```diff
@@ -79,9 +94,135 @@
...
+         */
+        eventCtrl.goToEvent = (event) => {
+            $state.go('app.user.event', { eventKey: event.key });
+        };
+
+        /**
+         * Get the color of institutional profile of the user
+         * @param {object} event - The current event
+         */
+        eventCtrl.getProfileColor = (event) => {
```
Comment (JuliePessoa) [inline frontend/event/eventController.js]: teste

File: frontend/event/eventController.js [hunk 25]
```diff
@@ -2,26 +2,33 @@
...
-
+        eventCtrl.eventsByDay = [];
+        eventCtrl.months = [];
+        eventCtrl.years = [];
+        eventCtrl.selectedMonth = null;
+        eventCtrl.selectedYear = null;
         eventCtrl.user = AuthService.getCurrentUser();
         eventCtrl.isLoadingEvents = true;
 
         eventCtrl.loadMoreEvents = function loadMoreEvents() {
```
Comment (JuliePessoa) [inline frontend/event/eventController.js]: teste

File: frontend/event/eventController.js [hunk 26]
```diff
@@ -32,16 +39,22 @@
         Utils.setScrollListener(content, eventCtrl.loadMoreEvents);
 
 
-        function loadEvents(deferred, getEvents) {
-            getEvents(actualPage, eventCtrl.institutionKey).then(function success(response) {
+        function loadEvents(deferred, getEvents, month, year) {
```
Comment (JuliePessoa) [inline frontend/event/eventController.js]: teste

File: backend/handlers/event_collection_handler.py [hunk 27]
```diff
@@ -39,20 +52,19 @@ def get_filtered_events(filters, user):
...
         if query.count() > 0:
             return ndb.gql("SELECT * FROM Event WHERE __key__ IN :1 AND end_time >= DATETIME(:2)",
-                query.fetch(), current_date.strftime("%Y-%m-%d %H:%M:%S")).order(Event.start_time, Event.key)
-        return query.order(Event.start_time, Event.key)
+                query.fetch(), begin_selected_month_utc.strftime("%Y-%m-%d %H:%M:%S"))
+        return query
     else:
         return Event.query(Event.institution_key.IN(
-            user.follows), Event.state == 'published').order(Event.start_time, Event.key)
+            user.follows), Event.state == 'published')
```
Comment (JuliePessoa) [inline backend/handlers/event_collection_handler.py]: why did you remove this order?

File: frontend/event/eventController.js [hunk 28]
```diff
@@ -40,7 +40,8 @@
 
 
         function loadEvents(deferred, getEvents, month, year) {
-            getEvents(actualPage, month, year, eventCtrl.institutionKey).then(function success(response) {
+            getEvents(actualPage, _.get(eventCtrl, 'institutionKey', month),
```
Comment (JuliePessoa) [inline frontend/event/eventController.js]: it gets very confusing.
You can use named parameters. So you pass an object as a parameter.
So it would be something like 
{actualPage, institutionKey: eventCtrl.institutionKey, month, year}
and in the function assignment you would expect
{actualPage, institutionKey, month, year}

File: frontend/event/events_mobile.html [hunk 29]
```diff
@@ -31,8 +31,9 @@ <h4>Nenhum evento a ser exibido.</h4>
               {{events.events[0].start_time | amUtc | amDateFormat:'ddd' }}
             </p>
           </div>
-          <div ng-repeat="event in events.events | orderBy:'start_time'" class="event-position">
-            <div ng-click="eventCtrl.goToEvent(event)">
+          <div ng-repeat="event in events.events | orderBy:'start_time'" class="event-position"
+            ng-click="eventCtrl.goToEvent(event)">
+            <div>
```
Comment (JuliePessoa) [inline frontend/event/events_mobile.html]: why a `div` tag with nothing ?

File: frontend/test/specs/event/eventControllerSpec.js [hunk 30]
```diff
@@ -91,66 +100,245 @@
...
+            state.params.institutionKey = institution.key;
+            eventCtrl.$onInit();
+            expect(eventCtrl.institutionKey).toEqual(institution.key);
+        });
+
+        it("Should call _getMonths() if is mobile screen", () => {
+            spyOn(Utils, 'isMobileScreen').and.returnValue(true);
+            spyOn(eventCtrl, '_getMonths');
+            eventCtrl.$onInit();
+            expect(eventCtrl._getMonths).toHaveBeenCalled();
```
Comment (JuliePessoa) [inline frontend/test/specs/event/eventControllerSpec.js]: verify that loadMoreEvents was not called.

File: frontend/test/specs/event/eventControllerSpec.js [hunk 31]
```diff
@@ -91,66 +100,245 @@
...
+        });
+
+        it("Should call loadMoreEvents() if is not mobile screen", () => {
+            spyOn(Utils, 'isMobileScreen').and.returnValue(false);
+            spyOn(eventCtrl, 'loadMoreEvents');
+            eventCtrl.$onInit();
+            expect(eventCtrl.loadMoreEvents).toHaveBeenCalled();
```
Comment (JuliePessoa) [inline frontend/test/specs/event/eventControllerSpec.js]: verify that getMonths was not called.

File: frontend/test/specs/event/eventControllerSpec.js [hunk 32]
```diff
@@ -91,66 +100,245 @@
...
+        });
+    });
+
+    describe('loadFilteredEvents()', () => {
+
+        beforeEach(() => {
+            spyOn(eventCtrl, 'loadMoreEvents');
+        });
+
+        it('Should call loadMoreEvents', () => {
```
Comment (JuliePessoa) [inline frontend/test/specs/event/eventControllerSpec.js]: theres another test for loadFilteredEvents at line 240. so remove this

File: frontend/test/specs/event/eventControllerSpec.js [hunk 33]
```diff
@@ -91,66 +100,245 @@
...
+        });
+    });
+
+    describe('loadMoreEvents()', () => {
+
+        it('Should call _loadEvents', () => {
+            spyOn(eventCtrl, '_loadEvents');
+            eventCtrl._moreEvents = true;
+            eventCtrl.loadMoreEvents();
+            expect(eventCtrl._loadEvents).toHaveBeenCalled();
```
Comment (JuliePessoa) [inline frontend/test/specs/event/eventControllerSpec.js]: check if it was called with the rigth getEventFunction
